src/main/java/bookstore/controller/BookController.java

```diff
@@ -0,0 +1,78 @@
+package bookstore.controller;
+
+import bookstore.dto.BookDto;
+import bookstore.dto.BookSearchParametersDto;
+import bookstore.dto.CreateBookRequestDto;
+import bookstore.service.BookService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Book management", description = "Endpoints for managing books")
+@RequiredArgsConstructor
+@RestController
+@Validated
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @GetMapping
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
```

Comment:
it may be beneficital to use `@PageableDefault` in case client did not provide pagination parameters